static/js/components/CommentForm.js

```diff
@@ -0,0 +1,223 @@
+    comment,
+    closeReply,
+    patchComment,
+    editing,
+    autoFocus
+  } = props
+
+  const [popupVisible, setPopupVisible] = useState(false)
+
+  return (
+    <div className={comment ? "reply-form" : "reply-post-form"}>
+      {userIsAnonymous() ? (
+        <LoginPopup
+          visible={popupVisible}
+          closePopup={setPopupVisible}
+          className="post-reply-popup"
+        />
+      ) : null}
+      <Formik
+        onSubmit={async (values, actions) => {
+          const { commentText } = values
+
+          if (editing && comment) {
+            const { id } = comment
+            const updatedComment = {
+              id,
+              text: commentText
+            }
+            await patchComment(updatedComment)
+          } else {
+            await onSubmit(commentText, comment ? comment.id : undefined, post)
+          }
+          actions.setSubmitting(false)
+          if (closeReply) {
+            closeReply()
+          } else {
+            actions.resetForm({ commentText: "" })
+          }
+        }}
+        initialValues={{
+          commentText: editing && comment ? comment.text : ""
+        }}
+      >
+        {({ isSubmitting, values, handleSubmit }) => (
+          <Form
+            onKeyDown={e => {
+              if (
+                e.key === "Enter" &&
+                e.ctrlKey &&
+                !isSubmitting &&
+                !isEmptyText(values.commentText)
+              ) {
+                handleSubmit(e)
+              }
+            }}
+          >
+            <div className="form-item">
+              {profile ? (
+                <React.Fragment>
+                  <ProfileImage
+                    profile={profile}
+                    imageSize={PROFILE_IMAGE_MICRO}
+                  />
+                  <div className="triangle" />
+                </React.Fragment>
+              ) : null}
+              <Field
+                type="text"
+                name="commentText"
+                className="input"
+                placeholder="Write a reply here..."
+                autoFocus={autoFocus}
+                render={({ field }) => (
+                  <textarea
+                    type="text"
+                    {...field}
+                    onChange={userOrAnonymousFunction(field.onChange, () =>
+                      setPopupVisible(true)
+                    )}
+                    onFocus={userOrAnonymousFunction(null, () =>
+                      setPopupVisible(true)
+                    )}
+                  />
+                )}
+              />
+            </div>
+            <button
+              type="submit"
+              disabled={
+                isSubmitting || userIsAnonymous() || !values.commentText
+              }
+            >
+              Submit
+            </button>
+            {closeReply ? (
+              <button
+                onClick={preventDefaultAndInvoke(closeReply)}
+                className="cancel"
```

Comment:
Should this be disabled when `isSubmitting` too?

Reply:
yeah, that makes sense I think